Sources/ArcGISToolkit/Components/Popups/PopupView.swift

```diff
@@ -126,13 +126,37 @@ public struct PopupView: View {
             }
         }
         .task(id: ObjectIdentifier(popup)) {
-            evaluateExpressionsResult = nil
-            evaluateExpressionsResult = await Result {
-                try await popup.evaluateExpressions()
+            // Initial evaluation for a newly assigned popup.
+            evaluation = nil
+            await evaluateExpressions()
+        }
+        .task(id: ObjectIdentifier(popup)) {
+            // If the popup is showing for a dynamic entity, then observe
+            // the changes and update the popup accordingly.
+            guard let dynamicEntity = popup.geoElement as? DynamicEntity else { return }
+            for await changes in dynamicEntity.changes {
+                if changes.dynamicEntityWasPurged {
+                    break
+                }
```

Comment:
Just for my own information, what happens to the popup/feature on the map when the Dynamic Entity is purged?

Reply:
it would simply stop updating the popup

Reply:
Does the popup get dismissed or the feature representing the dynamic entity get removed?

Reply:
The popup is not dismissed.

I believe that purged means that it was removed from rendering because it was purged from the datasource.